Automattic/woocommerce-payments
# Fix changing payment from saved payment for subscriptions order.

Fixes #1185

#### Changes proposed in this Pull Request

- Add a flag in Payment_Information to mark whether we are changing payment for subscriptions order or not.
- The bug is because when changing payment method using saved payment, we end up [in this if block](https://github.com/Automattic/woocommerce-payments/pull/1288/files#diff-bcfd303d507c33a27671935e724ee64d0df30ff2a18bb6ef4c94f28d48190124R575) and `add_token_to_order()` is not called, so we call `add_token_to_order()` and add order note to fix.









#### Testing instructions





Shamelessly copied from [1185](https://github.com/Automattic/woocommerce-payments/issues/1185).
1. Purchase subscription product.
2. Go to My account > Add one more payment method.
3. Go to My account > Subscription > Purchased subscription.
4. Try to change payment method to saved on step 2.
5. Success message is displayed but payment method is not changed. With this branch, payment method should be changed successfully.

-------------------
